src/Microsoft.Extensions.DependencyInjection.Abstractions/IDesignTimeServiceProviderFactory.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// A factory for creating the application's <see cref="IServiceProvider"/> at design time. Implement this interface
+    /// to enable design-time services that need access to the application's services. Design-time services will
+    /// automatically discover implementations of this interface.
```

Comment:
automatically discover implementations of this interface _in the startup assembly_?